File: Backend/airport-backend/src/main/java/vn/iuhcm/airport/ticket/Boarding.java
package vn.iuhcm.airport.ticket;

import java.time.LocalDateTime;

public class Boarding {

    public String passID; // Primary Key: UUID (VARCHAR(36))
    public String flightID;
    public String gateNum;
    public LocalDateTime boardingTime;
    public String seat;
    public String status; // 'Boarded','Waiting','Gate-Closed','No-Show'
    public String ticketID; // Unique Foreign Key to Ticket

    // Constructor rỗng (Jackson cần)
    public Boarding() {
    }

    // Constructor đầy đủ
    public Boarding(String passID, String flightID, String gateNum,
            LocalDateTime boardingTime, String seat, String status, String ticketID) {
        this.passID = passID;
        this.flightID = flightID;
        this.gateNum = gateNum;
        this.boardingTime = boardingTime;
        this.seat = seat;
        this.status = status;
        this.ticketID = ticketID;
    }
}

File: Frontend/airport-frontend/app/types/Boarding.ts
// File: app/types/Boarding.ts

export type BoardingStatus = 
  | 'Boarded' 
  | 'Waiting' 
  | 'Gate-Closed' 
  | 'No-Show';

export interface Boarding {
  /** Primary Key: UUID string (VARCHAR(36)) */
  passID: string;

  /** Foreign Key to Flight */
  flightID: string;

  /** Gate number */
  gateNum: string;

  /** DATETIME2 in ISO string format */
  boardingTime: string;

  /** Seat number (VARCHAR(10)) */
  seat: string;

  /** Status must be one of the defined BoardingStatus types */
  status: BoardingStatus;

  /** Unique Foreign Key to Ticket */
  ticketID: string;
}